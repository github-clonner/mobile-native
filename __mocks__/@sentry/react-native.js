export const captureMessage = jest.fn();
export default {};
