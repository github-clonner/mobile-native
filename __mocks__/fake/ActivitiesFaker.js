
function load(count, container) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      rowKey: 'something' + i,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      time_created: "1522036284",
      ownerObj:{
        guid: "824853017709780997",
        type: "user",
        subtype: false,
        time_created: "1522036284",
        // getAvatarSource: () => {
        //   return {
        //     source:'http://thisisaurl'
        //   }
        // }
      },
      shouldBeBlured: jest.fn(),
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
      wire_totals: {
        tokens: 1000000000000000000
      },
      containerObj: container
      // _list: {
      //   viewed: {
      //     viewed: new Map([["1019155171608096768",true]]),
      //     addViewed: () => {
      //       return;
      //     }
      //   }
      // },
      // getThumbSource: () => {
      //   return {
      //     source:'http://thisisaurl'
      //   }
      // }
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }

}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load}
}

