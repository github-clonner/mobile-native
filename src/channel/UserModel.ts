import { observable, action, runInAction } from 'mobx';

import { MINDS_CDN_URI, GOOGLE_PLAY_STORE } from '../config/Config';
import api from '../common/services/api.service';
import BaseModel from '../common/BaseModel';
import ChannelService from './ChannelService';
import sessionService from '../common/services/session.service';
import apiService from '../common/services/api.service';
import logService from '../common/services/log.service';

//@ts-nocheck
export const USER_MODE_OPEN = 0;
export const USER_MODE_MODERATED = 1;
export const USER_MODE_CLOSED = 2;

/**
 * User model
 */
export default class UserModel extends BaseModel {
  merchant;
  /**
   * Eth wallet
   */
  eth_wallet: string = '';
  wire_rewards;
  sums;
  btc_address?: string;
  icontime!: string;
  username!: string;
  name!: string;
  admin = false;
  plus: boolean = false;
  verified: boolean = false;
  founder: boolean = false;
  rewards: boolean = false;
  last_accepted_tos: number = 0;
  carousels?: Array<any>;

  /**
   * @var {boolean}
   */
  @observable blocked;

  /**
   * @var {number}
   */
  @observable subscribers_count;

  /**
   * @var {number}
   */
  @observable impressions;

  /**
   * @var {boolean}
   */
  @observable subscribed;

  /**
   * @var {boolean}
   */
  @observable mature_visibility = false;

  /**
   * @var {boolean}
   */
  @observable pending_subscribe = false;

  /**
   * @var {numeric}
   */
  @observable mode = 0;

  /**
   * @var {boolean}
   */
  @observable email_confirmed = false;

  /**
   * Confirm email
   * @param {Object} params
   */
  confirmEmail = async (params) => {
    // call any api endpoint with the param
    try {
      await apiService.get('api/v2/entities/', { urn: this.urn, ...params });
      this.setEmailConfirmed(true);
      return true;
    } catch (error) {
      return false;
    }
  };

  /**
   * Get the user icon time
   */
  getOwnerIcontime(): string {
    if (sessionService.getUser().guid === this.guid) {
      return sessionService.getUser().icontime;
    } else {
      return this.icontime;
    }
  }

  @action
  toggleMatureVisibility() {
    if (GOOGLE_PLAY_STORE) return;
    this.mature_visibility = !this.mature_visibility;
  }

  @action
  async toggleSubscription() {
    const value = !this.subscribed;
    this.subscribed = value;
    try {
      const metadata = this.getClientMetadata();
      await ChannelService.toggleSubscription(this.guid, value, metadata);
      UserModel.events.emit('toggleSubscription', this);
    } catch (err) {
      runInAction(() => {
        this.subscribed = !value;
      });
      throw err;
    }
  }

  @action
  async toggleBlock(value: boolean | null = null) {
    value = value === null ? !this.blocked : value;

    try {
      await ChannelService.toggleBlock(this.guid, value);
      this.blocked = value;
    } catch (err) {
      this.blocked = !value;
      logService.exception('[ChannelStore] toggleBlock', err);
    }
  }

  @action
  setMode(value) {
    this.mode = value;
  }

  @action
  setEmailConfirmed(value) {
    this.email_confirmed = value;
  }

  /**
   * Is admin
   */
  isAdmin() {
    return this.admin;
  }

  /**
   * current user is owner of the channel
   */
  isOwner = () => {
    return sessionService.getUser().guid === this.guid;
  };

  /**
   * Get banner source
   */
  getBannerSource() {
    if (this.carousels) {
      return {
        uri: this.carousels[0].src,
      };
    }
    return {
      uri: `${MINDS_CDN_URI}fs/v1/banners/${this.guid}/fat/${this.icontime}`,
      headers: api.buildHeaders(),
    };
  }

  /**
   * Get avatar source
   * @param {string} size
   */
  getAvatarSource(size = 'medium') {
    return {
      uri: `${MINDS_CDN_URI}icon/${
        this.guid
      }/${size}/${this.getOwnerIcontime()}`,
      headers: api.buildHeaders(),
    };
  }

  /**
   * Has banner
   * @returns {boolean}
   */
  hasBanner() {
    return !!this.carousels;
  }

  /**
   * Is closed
   * @returns {boolean}
   */
  isClosed() {
    return this.mode === USER_MODE_CLOSED;
  }

  /**
   * Is open
   * @returns {boolean}
   */
  isOpen() {
    return this.mode === USER_MODE_OPEN;
  }

  /**
   * Is moderated
   * @returns {boolean}
   */
  isModerated() {
    return this.mode === USER_MODE_MODERATED;
  }

  /**
   * Is subscribed
   * @returns {boolean}
   */
  isSubscribed() {
    return !!this.subscribed;
  }

  /**
   * Request subscribe
   */
  async subscribeRequest() {
    if (this.pending_subscribe || this.mode !== USER_MODE_CLOSED) {
      return;
    }
    try {
      this.pending_subscribe = true;
      await apiService.put(`api/v2/subscriptions/outgoing/${this.guid}`);
    } catch (err) {
      this.pending_subscribe = false;
      logService.exception(err);
    }
  }

  /**
   * Cancel subscribe request
   */
  async cancelSubscribeRequest() {
    if (!this.pending_subscribe || this.mode !== USER_MODE_CLOSED) {
      return;
    }
    try {
      this.pending_subscribe = false;
      await apiService.delete(`api/v2/subscriptions/outgoing/${this.guid}`);
    } catch (err) {
      this.pending_subscribe = true;
      logService.exception(err);
    }
  }
}
