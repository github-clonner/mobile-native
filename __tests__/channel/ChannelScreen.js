import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import ChannelScreen from '../../src/channel/ChannelScreen';
import { activitiesServiceFaker } from '../../__mocks__/fake/ActivitiesFaker';
import userFaker from '../../__mocks__/fake/channel/UserFactory';
import UserModel, {
  USER_MODE_CLOSED,
  USER_MODE_OPEN,
} from '../../src/channel/UserModel';
import featuresService from '../../src/common/services/features.service';
import session from '../../src/common/services/session.service';

jest.mock('../../src/common/helpers/abortableFetch');
jest.mock('../../src/newsfeed/ActivityModel');
jest.mock(
  '../../src/channel/carousel/RewardsCarousel',
  () => 'RewardsCarousel',
);
jest.mock('../../src/channel/header/ChannelHeader', () => 'ChannelHeader');
jest.mock('../../src/channel/toolbar/Toolbar', () => 'Toolbar');
jest.mock(
  '../../src/common/components/CenteredLoading',
  () => 'CenteredLoading',
);
jest.mock('../../src/common/components/Button', () => 'Button');
jest.mock('../../src/common/components/FeedList', () => 'FeedList');
jest.mock('../../src/capture/CaptureFab', () => 'CaptureFab');
jest.mock('../../src/blogs/BlogCard', () => 'BlogCard');
jest.mock('../../src/common/components/Touchable', () => 'Touchable');
jest.mock('../../src/common/services/boosted-content.service');
jest.mock('../../src/common/services/session.service');

/**
 * Tests
 */
describe('Channel screen component', () => {
  let screen, channel, navigation, user, channelStore, route;

  session.getUser.mockReturnValue(UserModel.create(userFaker(1)));

  beforeEach(() => {
    let activityResponse = activitiesServiceFaker().load(1);
    let activity = activityResponse.activities[0];
    activity.wire_totals = { tokens: 20 };
    activity.impressions = 20;

    navigation = { navigate: jest.fn(), goBack: jest.fn() };
    route = {
      params: {
        entity: activity,
      },
    };
    channelStore = {
      channel: UserModel.create(userFaker(1)),
      setChannel: jest.fn(),
      feedStore: {
        setChannel: jest.fn(),
      },
      rewards: {
        merged: [],
      },
      load: jest.fn(),
    };

    channel = { store: jest.fn() };

    channel.store.mockReturnValue(channelStore);

    screen = shallow(
      <ChannelScreen.wrappedComponent
        channel={channel}
        navigation={navigation}
        route={route}
      />,
    );
  });

  it('should renders correctly', () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });

  it('should have the expected components while loading, also check newsfeed', async () => {
    expect(screen.find('FeedList')).toHaveLength(1);
    let render = screen.dive();
    expect(render.find('CenteredLoading')).toHaveLength(0);
    expect(screen.find('CaptureFab')).toHaveLength(1);
  });

  it('should have the expected components while loading, also check flatlist for blocked', async () => {
    channelStore.channel.blocked = true;

    await screen.update();

    expect(screen.find('FeedList')).toHaveLength(0);
    let render = screen.dive();
    expect(render.find('CenteredLoading')).toHaveLength(0);
    expect(screen.find('CaptureFab')).toHaveLength(1);
  });

  it('should show closed channel message', async () => {
    channelStore.channel.blocked = false;
    channelStore.channel.mode = USER_MODE_CLOSED;

    await screen.update();

    expect(screen).toMatchSnapshot();
  });

  it('should navigate back if the the VIEW permissions are removed', async () => {
    channelStore.channel.mode = USER_MODE_OPEN;

    // force permissions feature flag
    featuresService.features = { permissions: true };

    channelStore.channel.setPermissions({ permissions: [] });

    await screen.update();

    expect(navigation.goBack).toBeCalled();
  });
});
