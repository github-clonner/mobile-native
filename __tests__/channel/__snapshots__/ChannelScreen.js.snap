// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel screen component should renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="ChannelScreen"
>
  <FeedList
    emptyMessage={null}
    header={
      <View>
        <ChannelHeader
          navigation={
            Object {
              "goBack": [MockFunction],
              "navigate": [MockFunction],
            }
          }
          store={
            Object {
              "channel": UserModel {
                "__list": null,
                "access_id": "2",
                "admin": false,
                "allow_comments": true,
                "banned": "no",
                "blocked": false,
                "boostProPlus": false,
                "boost_autorotate": true,
                "boost_rating": "2",
                "briefdescription": "",
                "categories": Array [
                  "music",
                  "nature",
                  "news",
                  "politics",
                  "science",
                  "technology",
                  "education",
                  "art",
                  "health",
                ],
                "chat": true,
                "city": "",
                "confirmEmail": [Function],
                "container_guid": "0",
                "deleted": "0",
                "disabled_boost": false,
                "dob": "",
                "email_confirmed": false,
                "eth_wallet": "0x6fffffdadae350ddd5a1777d5a1777a4f39f0317",
                "fb": false,
                "feature_flags": false,
                "featured_id": "725165605782560768",
                "founder": false,
                "gender": "private",
                "getAvatarSource": [MockFunction],
                "guid": 1,
                "icontime": "1523515420",
                "impressions": 100,
                "isOwner": [Function],
                "language": "en",
                "last_accepted_tos": 0,
                "legacy_guid": false,
                "mature": "1",
                "mature_channel": "0",
                "mature_visibility": false,
                "merchant": Object {
                  "exclusive": Object {
                    "amount": "10",
                    "enabled": true,
                  },
                  "id": "acct_Dsdfjdlsaf34243",
                  "service": "stripe",
                },
                "mode": 0,
                "monetized": "",
                "name": "Some User",
                "owner_guid": "0",
                "pending_subscribe": false,
                "permissions": Object {},
                "pinned_posts": Array [],
                "plus": true,
                "programs": Array [
                  "affiliate",
                ],
                "rating": "1",
                "rewards": true,
                "signup_method": false,
                "site_guid": false,
                "social_profiles": Array [],
                "spam": "0",
                "subscribed": false,
                "subscriber": true,
                "subscribers_count": 50,
                "subtype": false,
                "time_created": "1468113204",
                "time_updated": false,
                "type": "user",
                "urn": "",
                "username": "someusermane",
                "verified": true,
                "website": "",
                "wire_rewards": Object {
                  "description": "",
                  "rewards": Object {
                    "money": Array [],
                    "points": Array [],
                  },
                },
              },
              "feedStore": Object {
                "setChannel": [MockFunction],
              },
              "load": [MockFunction],
              "rewards": Object {
                "merged": Array [],
              },
              "setChannel": [MockFunction],
            }
          }
          styles={
            Object {
              "avatar": Object {
                "borderRadius": 55,
                "height": 110,
                "left": 20,
                "position": "absolute",
                "top": 135,
                "width": 110,
              },
              "banner": Object {
                "alignItems": "stretch",
                "backgroundColor": "#eee",
                "flexDirection": "row",
                "height": 190,
                "width": "100%",
              },
              "blockText": Object {
                "fontFamily": "Roboto",
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 1,
                "padding": 20,
                "paddingTop": 0,
              },
              "blockTextLink": Object {
                "color": "#4690D6",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "fontWeight": "400",
                "letterSpacing": 1,
                "padding": 20,
                "paddingTop": 0,
              },
              "blockView": Object {
                "alignItems": "center",
              },
              "bluebutton": Object {
                "marginRight": 8,
              },
              "briefdescription": Object {
                "color": "#777",
                "fontFamily": "Roboto",
                "fontSize": 13,
                "paddingBottom": 20,
                "paddingTop": 25,
              },
              "briefdescriptionTextInput": Object {
                "color": "#777",
                "fontFamily": "Roboto",
                "fontSize": 13,
                "maxHeight": 100,
              },
              "briefdescriptionTextInputView": Object {
                "borderColor": "#ececec",
                "borderRadius": 3,
                "borderWidth": 1,
                "marginBottom": 20,
                "marginTop": 20,
                "padding": 8,
                "paddingTop": 3,
              },
              "buttonscol": Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
              "carouselcontainer": Object {
                "flex": 1,
                "paddingBottom": 20,
              },
              "container": Object {
                "backgroundColor": "#FFF",
                "flex": 1,
                "flexDirection": "column",
              },
              "counter": Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
              },
              "countercontainer": Object {
                "flexDirection": "row",
                "height": 60,
                "paddingLeft": 130,
              },
              "countertitle": Object {
                "color": "#666",
                "fontSize": 10,
              },
              "countervalue": Object {
                "color": "#444",
                "fontWeight": "bold",
                "paddingTop": 5,
              },
              "gobackicon": Object {
                "left": 0,
                "position": "absolute",
                "top": 16,
              },
              "headercontainer": Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 200,
              },
              "headertextcontainer": Object {
                "alignItems": "stretch",
                "flexDirection": "column",
                "padding": 8,
                "paddingLeft": 15,
                "paddingRight": 15,
                "width": "100%",
              },
              "name": Object {
                "color": "#444",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "700",
                "letterSpacing": 0.5,
                "marginRight": 8,
              },
              "nameTextInput": Object {
                "borderColor": "#ececec",
                "borderRadius": 3,
                "borderWidth": 1,
                "color": "#444",
                "fontFamily": "Roboto",
                "fontSize": 20,
                "fontWeight": "700",
                "letterSpacing": 0.5,
                "marginBottom": 4,
                "padding": 3,
              },
              "namecol": Object {
                "flex": 1,
              },
              "namecontainer": Object {
                "flex": 1,
                "flexDirection": "row",
              },
              "tapOverlayView": Object {
                "alignItems": "center",
                "backgroundColor": "#000",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 0.65,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              "username": Object {
                "color": "#999",
                "fontSize": 14,
              },
              "wrappedAvatar": Object {
                "borderRadius": 55,
                "height": 110,
                "width": 110,
              },
              "wrappedAvatarOverlayView": Object {
                "borderRadius": 55,
              },
            }
          }
        />
        <Toolbar
          channel={
            UserModel {
              "__list": null,
              "access_id": "2",
              "admin": false,
              "allow_comments": true,
              "banned": "no",
              "blocked": false,
              "boostProPlus": false,
              "boost_autorotate": true,
              "boost_rating": "2",
              "briefdescription": "",
              "categories": Array [
                "music",
                "nature",
                "news",
                "politics",
                "science",
                "technology",
                "education",
                "art",
                "health",
              ],
              "chat": true,
              "city": "",
              "confirmEmail": [Function],
              "container_guid": "0",
              "deleted": "0",
              "disabled_boost": false,
              "dob": "",
              "email_confirmed": false,
              "eth_wallet": "0x6fffffdadae350ddd5a1777d5a1777a4f39f0317",
              "fb": false,
              "feature_flags": false,
              "featured_id": "725165605782560768",
              "founder": false,
              "gender": "private",
              "getAvatarSource": [MockFunction],
              "guid": 1,
              "icontime": "1523515420",
              "impressions": 100,
              "isOwner": [Function],
              "language": "en",
              "last_accepted_tos": 0,
              "legacy_guid": false,
              "mature": "1",
              "mature_channel": "0",
              "mature_visibility": false,
              "merchant": Object {
                "exclusive": Object {
                  "amount": "10",
                  "enabled": true,
                },
                "id": "acct_Dsdfjdlsaf34243",
                "service": "stripe",
              },
              "mode": 0,
              "monetized": "",
              "name": "Some User",
              "owner_guid": "0",
              "pending_subscribe": false,
              "permissions": Object {},
              "pinned_posts": Array [],
              "plus": true,
              "programs": Array [
                "affiliate",
              ],
              "rating": "1",
              "rewards": true,
              "signup_method": false,
              "site_guid": false,
              "social_profiles": Array [],
              "spam": "0",
              "subscribed": false,
              "subscriber": true,
              "subscribers_count": 50,
              "subtype": false,
              "time_created": "1468113204",
              "time_updated": false,
              "type": "user",
              "urn": "",
              "username": "someusermane",
              "verified": true,
              "website": "",
              "wire_rewards": Object {
                "description": "",
                "rewards": Object {
                  "money": Array [],
                  "points": Array [],
                },
              },
            }
          }
          feed={
            Object {
              "setChannel": [MockFunction],
            }
          }
          hasRewards={0}
        />
      </View>
    }
    navigation={
      Object {
        "goBack": [MockFunction],
        "navigate": [MockFunction],
      }
    }
    renderActivity={null}
  />
  <SafeAreaView
    style={
      Object {
        "left": 0,
        "position": "absolute",
        "top": 16,
      }
    }
  >
    <Themed.Icon
      color="#4690D6"
      name="arrow-back"
      onPress={[Function]}
      raised={true}
      size={22}
    />
  </SafeAreaView>
  <CaptureFab
    navigation={
      Object {
        "goBack": [MockFunction],
        "navigate": [MockFunction],
      }
    }
    route={
      Object {
        "params": Object {
          "entity": Object {
            "attachment_guid": false,
            "blurb": false,
            "containerObj": undefined,
            "container_guid": "activityguid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "activityguid0",
            "impressions": 20,
            "mature": false,
            "message": "Message",
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "rowKey": "something0",
            "shouldBeBlured": [MockFunction],
            "thumbnail_src": false,
            "time_created": "1522036284",
            "title": "TITLE",
            "type": "activity",
            "wire_totals": Object {
              "tokens": 20,
            },
          },
        },
      }
    }
    testID="captureFab"
  />
</View>
`;

exports[`Channel screen component should show closed channel message 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="ChannelScreen"
>
  <FeedList
    emptyMessage={null}
    header={
      <View>
        <ChannelHeader
          navigation={
            Object {
              "goBack": [MockFunction],
              "navigate": [MockFunction],
            }
          }
          store={
            Object {
              "channel": UserModel {
                "__list": null,
                "access_id": "2",
                "admin": false,
                "allow_comments": true,
                "banned": "no",
                "blocked": false,
                "boostProPlus": false,
                "boost_autorotate": true,
                "boost_rating": "2",
                "briefdescription": "",
                "categories": Array [
                  "music",
                  "nature",
                  "news",
                  "politics",
                  "science",
                  "technology",
                  "education",
                  "art",
                  "health",
                ],
                "chat": true,
                "city": "",
                "confirmEmail": [Function],
                "container_guid": "0",
                "deleted": "0",
                "disabled_boost": false,
                "dob": "",
                "email_confirmed": false,
                "eth_wallet": "0x6fffffdadae350ddd5a1777d5a1777a4f39f0317",
                "fb": false,
                "feature_flags": false,
                "featured_id": "725165605782560768",
                "founder": false,
                "gender": "private",
                "getAvatarSource": [MockFunction],
                "guid": 1,
                "icontime": "1523515420",
                "impressions": 100,
                "isOwner": [Function],
                "language": "en",
                "last_accepted_tos": 0,
                "legacy_guid": false,
                "mature": "1",
                "mature_channel": "0",
                "mature_visibility": false,
                "merchant": Object {
                  "exclusive": Object {
                    "amount": "10",
                    "enabled": true,
                  },
                  "id": "acct_Dsdfjdlsaf34243",
                  "service": "stripe",
                },
                "mode": 2,
                "monetized": "",
                "name": "Some User",
                "owner_guid": "0",
                "pending_subscribe": false,
                "permissions": Object {},
                "pinned_posts": Array [],
                "plus": true,
                "programs": Array [
                  "affiliate",
                ],
                "rating": "1",
                "rewards": true,
                "signup_method": false,
                "site_guid": false,
                "social_profiles": Array [],
                "spam": "0",
                "subscribed": false,
                "subscriber": true,
                "subscribers_count": 50,
                "subtype": false,
                "time_created": "1468113204",
                "time_updated": false,
                "type": "user",
                "urn": "",
                "username": "someusermane",
                "verified": true,
                "website": "",
                "wire_rewards": Object {
                  "description": "",
                  "rewards": Object {
                    "money": Array [],
                    "points": Array [],
                  },
                },
              },
              "feedStore": Object {
                "setChannel": [MockFunction],
              },
              "load": [MockFunction],
              "rewards": Object {
                "merged": Array [],
              },
              "setChannel": [MockFunction],
            }
          }
          styles={
            Object {
              "avatar": Object {
                "borderRadius": 55,
                "height": 110,
                "left": 20,
                "position": "absolute",
                "top": 135,
                "width": 110,
              },
              "banner": Object {
                "alignItems": "stretch",
                "backgroundColor": "#eee",
                "flexDirection": "row",
                "height": 190,
                "width": "100%",
              },
              "blockText": Object {
                "fontFamily": "Roboto",
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 1,
                "padding": 20,
                "paddingTop": 0,
              },
              "blockTextLink": Object {
                "color": "#4690D6",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "fontWeight": "400",
                "letterSpacing": 1,
                "padding": 20,
                "paddingTop": 0,
              },
              "blockView": Object {
                "alignItems": "center",
              },
              "bluebutton": Object {
                "marginRight": 8,
              },
              "briefdescription": Object {
                "color": "#777",
                "fontFamily": "Roboto",
                "fontSize": 13,
                "paddingBottom": 20,
                "paddingTop": 25,
              },
              "briefdescriptionTextInput": Object {
                "color": "#777",
                "fontFamily": "Roboto",
                "fontSize": 13,
                "maxHeight": 100,
              },
              "briefdescriptionTextInputView": Object {
                "borderColor": "#ececec",
                "borderRadius": 3,
                "borderWidth": 1,
                "marginBottom": 20,
                "marginTop": 20,
                "padding": 8,
                "paddingTop": 3,
              },
              "buttonscol": Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
              "carouselcontainer": Object {
                "flex": 1,
                "paddingBottom": 20,
              },
              "container": Object {
                "backgroundColor": "#FFF",
                "flex": 1,
                "flexDirection": "column",
              },
              "counter": Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
              },
              "countercontainer": Object {
                "flexDirection": "row",
                "height": 60,
                "paddingLeft": 130,
              },
              "countertitle": Object {
                "color": "#666",
                "fontSize": 10,
              },
              "countervalue": Object {
                "color": "#444",
                "fontWeight": "bold",
                "paddingTop": 5,
              },
              "gobackicon": Object {
                "left": 0,
                "position": "absolute",
                "top": 16,
              },
              "headercontainer": Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 200,
              },
              "headertextcontainer": Object {
                "alignItems": "stretch",
                "flexDirection": "column",
                "padding": 8,
                "paddingLeft": 15,
                "paddingRight": 15,
                "width": "100%",
              },
              "name": Object {
                "color": "#444",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "700",
                "letterSpacing": 0.5,
                "marginRight": 8,
              },
              "nameTextInput": Object {
                "borderColor": "#ececec",
                "borderRadius": 3,
                "borderWidth": 1,
                "color": "#444",
                "fontFamily": "Roboto",
                "fontSize": 20,
                "fontWeight": "700",
                "letterSpacing": 0.5,
                "marginBottom": 4,
                "padding": 3,
              },
              "namecol": Object {
                "flex": 1,
              },
              "namecontainer": Object {
                "flex": 1,
                "flexDirection": "row",
              },
              "tapOverlayView": Object {
                "alignItems": "center",
                "backgroundColor": "#000",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 0.65,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              "username": Object {
                "color": "#999",
                "fontSize": 14,
              },
              "wrappedAvatar": Object {
                "borderRadius": 55,
                "height": 110,
                "width": 110,
              },
              "wrappedAvatarOverlayView": Object {
                "borderRadius": 55,
              },
            }
          }
        />
        <Toolbar
          channel={
            UserModel {
              "__list": null,
              "access_id": "2",
              "admin": false,
              "allow_comments": true,
              "banned": "no",
              "blocked": false,
              "boostProPlus": false,
              "boost_autorotate": true,
              "boost_rating": "2",
              "briefdescription": "",
              "categories": Array [
                "music",
                "nature",
                "news",
                "politics",
                "science",
                "technology",
                "education",
                "art",
                "health",
              ],
              "chat": true,
              "city": "",
              "confirmEmail": [Function],
              "container_guid": "0",
              "deleted": "0",
              "disabled_boost": false,
              "dob": "",
              "email_confirmed": false,
              "eth_wallet": "0x6fffffdadae350ddd5a1777d5a1777a4f39f0317",
              "fb": false,
              "feature_flags": false,
              "featured_id": "725165605782560768",
              "founder": false,
              "gender": "private",
              "getAvatarSource": [MockFunction],
              "guid": 1,
              "icontime": "1523515420",
              "impressions": 100,
              "isOwner": [Function],
              "language": "en",
              "last_accepted_tos": 0,
              "legacy_guid": false,
              "mature": "1",
              "mature_channel": "0",
              "mature_visibility": false,
              "merchant": Object {
                "exclusive": Object {
                  "amount": "10",
                  "enabled": true,
                },
                "id": "acct_Dsdfjdlsaf34243",
                "service": "stripe",
              },
              "mode": 2,
              "monetized": "",
              "name": "Some User",
              "owner_guid": "0",
              "pending_subscribe": false,
              "permissions": Object {},
              "pinned_posts": Array [],
              "plus": true,
              "programs": Array [
                "affiliate",
              ],
              "rating": "1",
              "rewards": true,
              "signup_method": false,
              "site_guid": false,
              "social_profiles": Array [],
              "spam": "0",
              "subscribed": false,
              "subscriber": true,
              "subscribers_count": 50,
              "subtype": false,
              "time_created": "1468113204",
              "time_updated": false,
              "type": "user",
              "urn": "",
              "username": "someusermane",
              "verified": true,
              "website": "",
              "wire_rewards": Object {
                "description": "",
                "rewards": Object {
                  "money": Array [],
                  "points": Array [],
                },
              },
            }
          }
          feed={
            Object {
              "setChannel": [MockFunction],
            }
          }
          hasRewards={0}
        />
      </View>
    }
    navigation={
      Object {
        "goBack": [MockFunction],
        "navigate": [MockFunction],
      }
    }
    renderActivity={null}
  />
  <SafeAreaView
    style={
      Object {
        "left": 0,
        "position": "absolute",
        "top": 16,
      }
    }
  >
    <Themed.Icon
      color="#4690D6"
      name="arrow-back"
      onPress={[Function]}
      raised={true}
      size={22}
    />
  </SafeAreaView>
  <CaptureFab
    navigation={
      Object {
        "goBack": [MockFunction],
        "navigate": [MockFunction],
      }
    }
    route={
      Object {
        "params": Object {
          "entity": Object {
            "attachment_guid": false,
            "blurb": false,
            "containerObj": undefined,
            "container_guid": "activityguid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "activityguid0",
            "impressions": 20,
            "mature": false,
            "message": "Message",
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "rowKey": "something0",
            "shouldBeBlured": [MockFunction],
            "thumbnail_src": false,
            "time_created": "1522036284",
            "title": "TITLE",
            "type": "activity",
            "wire_totals": Object {
              "tokens": 20,
            },
          },
        },
      }
    }
    testID="captureFab"
  />
</View>
`;
